Countdown to the New Year: 31 Days of Ember Addons - ember-exam (Dec 30th 2019)

jayjayjpg opened the pull request:
<!--- Make sure to add a descriptive title in the field above! E.g. "Fixes the header title color on the homepage"  -->

## What it does
<!--- Tell us what this fix does in a few sentences. E.g. "This updates the header title's font color to Ember Orange." -->

When merged, this will create a new blog post for the DecEmber 2019 blog series.

Should be merged on December 30th, 2019.

## Related Issue(s)
<!--- Please provide the issue(s) to which this pull request relates to or which issue it closes. E.g. "Closes #1234" -->

## Sources
<!-- Optional. If applicable be sure to add any screenshots or screen recordings of your work for your reviewers here -->

jayjayjpg committed:
feat(blog:cttny#30): add writeup about ember exam (day 30)

MelSumner commented on source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md:
```diff
@@ -0,0 +1,21 @@
+---
+title: Countdown to The New Year: Ember Exam
+author: Jessica Jordan
+tags: 2019, Recent Posts, Testing
```
let's take out Recent Posts, so it doesn't bury the 3.15 release blog post- I wonder if we can have a different tag here- `DecEmber` maybe?

jayjayjpg replied:
Yes, that sounds great! I also used the tags from your PR #365 as a reference :+1:

MelSumner commented on source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md:
```diff
@@ -0,0 +1,21 @@
+
+Ember Exam allows us to run tests in parallel which can greatly reduce the overall run time of our test suite. Using the `--random` option of this addons, we may run tests in random order uncovering interdependencies between test cases which would be hard to spot otherwise. And the `exam:iterate` allows us to iterate over parts of our test suite repeatedly to track down brittle tests easily.
+
+Ember Exam helps us to develop a more robust, reliable and faster test suite - which makes our testing workflow in Ember more predictable, more relaxed and therefore also more fun! [Check it out](https://emberobserver.com/addons/ember-exam)!
```
"Check it out for yourself" would be an improvement for a user with a screen reader.

jayjayjpg replied:
That sounds great, thank you for the suggestion :sparkles:

MelSumner reviewed (changes requested):
Thank you for working on this, @jessica-jordan ! :+1:

jayjayjpg committed:
upd(blog:cttny#30): update post according to general format + review

jayjayjpg commented:
@MelSumner Thank you for the suggestions! I updated the PR accordingly and also updated the overall structure to be similar to the one you used in #365 :sparkles:

Turbo87 commented on source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md:
```diff
@@ -0,0 +1,27 @@
+
+### What It Does
+
+Ember Exam is an addon that allows to have more control over running your Ember QUnit (Ember's default testing framework) and Ember Mocha test suite. It extends from the default Ember CLI `test` command, allowing you run it as you would `ember test`.
```
```suggestion
Ember Exam is an addon that allows to have more control over running your ember-qunit (Ember's default testing framework) and ember-mocha test suite. It extends from the default Ember CLI `test` command, allowing you run it as you would `ember test`.
```

jayjayjpg replied:
I think it makes sense to stick to one casing in the writeup - do you think it makes more sense to go all title case or lower case (`ember-exam`/ `ember-qunit` vs `Ember Exam` / `Ember QUnit`)?

I prefer title case personally, but I think it's after all just important to keep it uniform. What do you think?

Turbo87 replied:
- https://github.com/emberjs/ember-qunit#ember-qunit
- https://github.com/emberjs/ember-mocha#ember-mocha

there is no mention of these alternative title case addon names. I personally prefer using the actual names instead of trying to titleize everything like we originally did in the addon blueprints. when the names are multiple words it can make it hard in prose to figure out what words are still part of the name and which ones are not.

jayjayjpg replied:
Thank you for the explanation and the readability of long names is also a good point I haven't thought about yet - I'll update the casing for this and the other writeups accordingly

jayjayjpg committed:
Update source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md

Co-Authored-By: Tobias Bieniek <tobias.bieniek@gmail.com>

jayjayjpg committed:
upd(blog:cttny#30): update casing for ember-exam

amyrlam commented on source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md:
```diff
@@ -0,0 +1,27 @@
+responsive: true
+---
+
+This is already the second to last edition of our December series - "Countdown to The New Year: 31 Days of Ember Addons".  In this series we are highlighting a new addon each day until the new year, and we hope you'll join us for the fun!
```
```suggestion
This is already the second to last edition of our December series - "Countdown to The New Year: 31 Days of Ember Addons". In this series we are highlighting a new addon each day until the new year, and we hope you'll join us for the fun!
```

^ extra space there

Look good! In more of a backend-focused test suite now, but I love ember-exam. I really like the EmberMap video on the [README](https://github.com/ember-cli/ember-exam) if you want to mention it, but feel free to ignore me as well

jayjayjpg replied:
That sounds like a great addition as well - thank you for suggesting it! I'll update this accordingly

jayjayjpg committed:
Update source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md

Co-Authored-By: Amy Lam <amy.lam@betterup.co>

amyrlam commented on source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md:
```diff
@@ -0,0 +1,27 @@
+On the 30th day of our **Countdown to The New Year** we want to feature an exciting Ember addon to make your **test suite** fun to work with: **ember-exam**!
+
+Learn more about [ember-exam on Ember Observer](https://emberobserver.com/addons/ember-exam).
+
```
add `READMORE`

jayjayjpg replied:
```suggestion

READMORE

```

amyrlam commented on source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md:
```diff
@@ -0,0 +1,27 @@
+
+### Why I Like It
+
+ember-exam allows us to run tests in parallel which can greatly reduce the overall run time of our test suite. Using the `--random` option of this addon, we are able to run tests in random order; this helps us to uncover interdependencies between test cases which would be hard to spot otherwise. And the `exam:iterate` allows us to iterate over parts of our test suite repeatedly when tracking down ...
```
```suggestion
ember-exam allows us to run tests in parallel which can greatly reduce the overall run time of our test suite. Using the `--random` option of this addon, we are able to run tests in random order; this helps us to uncover interdependencies between test cases which would be hard to spot otherwise. And the `exam:iterate` option allows us to iterate over parts of our test suite repeatedly when tracking down brittle tests.
```

amyrlam reviewed (approved):
👍

fwiw, you may be able to merge these in and they won't appear until the calendar day comes up... but i'm not 100% sure

jayjayjpg committed:
Update source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md

jayjayjpg committed:
Update source/2019-12-30-countdown-to-the-new-year-2019-ember-exam.md

Co-Authored-By: Amy Lam <amy.lam@betterup.co>

amyrlam committed:
Add copy edits

Follow-on to https://github.com/ember-learn/ember-blog/pull/435

jayjayjpg commented:
@amyrlam Good point, I hadn't considered this yet. I just had a look at [the draft article section of the middle man documentation](https://middlemanapp.com/basics/blogging/#draft-articles) and found the following info:

> Draft articles will only appear in development mode.
An article with a date that is in the future is also considered unpublished; if you use a cron job to regenerate your site on a regular basis, this can be used to automatically publish articles at a specified time. This behaviour can be disabled by setting publish_future_dated to true.

I think this would be possible if we had an automatic rebuild of the website daily, but I think we currently only deploy once master changes, e.g. by merging a PR. What are your thoughts?

MelSumner reviewed (approved):
Thank you for writing this @jessica-jordan :+1:
